response_operations_ui/views/admin/manage_user_accounts.py

```diff
@@ -80,6 +103,85 @@ def manage_user_accounts():
     )
 
 
+@admin_bp.route("/create-account", methods=["GET"])
+@login_required
+def get_create_account():
+    _verify_user_in_user_admin_group()
+    form = CreateAccountWithPermissionsForm()
+    return render_template("admin/user-create.html", form=form)
+
+
+@admin_bp.route("/create-account", methods=["POST"])
+@login_required
+def post_create_account():
+    _verify_user_in_user_admin_group()
+    form = CreateAccountWithPermissionsForm(request.form)
+
+    if not form.validate():
+        return render_template("admin/user-create.html", form=form)
+
+    try:
+        groups_from_uaa = get_groups()
+    except HTTPError:
+        flash("Failed to get groups, please try again", "error")
+        return render_template("admin/user-create.html", form=form)
+
+    email = form.email.data
+    user = create_user_account_with_random_password(email, form.first_name.data, form.last_name.data)
+    if user.get("error"):
+        flash(user.get("error"), "error")
+        return render_template("admin/user-create.html", form=form)
+
+    user_id = user["id"]
+    groups_in_form = form.get_uaa_permission_groups()
+
+    for group, is_ticked in form.data.items():
+        if group not in groups_in_form or not is_ticked:
+            continue
+        mapped_group = uaa_group_mapping[group]
+        group_details = next(item for item in groups_from_uaa["resources"] if item["displayName"] == mapped_group)
+        try:
+            add_group_membership(user_id, group_details["id"])
+        except HTTPError:
+            # If a group permission add fails then we just need to continue, but let the person creating the account
+            # that it won't have all the permissions they requested.  This is easily fixed by them granting the
+            # permission via the manage accounts screen
+            flash(
+                f"Failed to give the user the {group} permission. The account has still been created but the "
+                f"permission will need to be granted later",
+                "error",
+            )
+            continue
+
+    token = token_decoder.generate_token(user_id)
+    internal_url = current_app.config["RESPONSE_OPERATIONS_UI_URL"]
+    verification_url = f"{internal_url}{url_for('account_bp.get_verify_account', token=token)}"
+    if not current_app.config["SEND_EMAIL_TO_GOV_NOTIFY"]:
+        logger.info("Verification url for new user", verification_link=verification_url)
+
+    try:
+        NotifyController().request_to_notify(
+            email=email,
+            template_name="create_user_account",
```

Comment:
Might want to check this as I messed with the templates on _get_template_id to simulate someone deleting it from notify, it didn't give me an error

Reply:
We talked about this and it's fine.  If the value is bad then we can't do anything about it as the failure will happen in the cloud function that sends the request to gov notify.
If the key is missing, it'll raise a keyError that'll bubble all the way up and cause a 500 error.  It's a bit messy but this should never ever happen so I'm okay with letting it break messy.